src/NuGet.Core/NuGet.ProjectManagement/Projects/MSBuildNuGetProject.cs

```diff
@@ -53,6 +53,16 @@ public class MSBuildNuGetProject : NuGetProject
         /// </summary>
         public event EventHandler<PackageEventArgs> PackageReferenceRemoved;
 
+        /// <summary>
+        /// Event to be raised when batch processing of install/ uninstall packages starts at a project level
+        /// </summary>
+        public event EventHandler<PackageEventArgs> BatchStart;
```

Comment:
I'm not sure if i follow it correctly but these events are only applicable for msbuild projects since this perf overhead is only there. Why would we need it for project.json?